File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.cpp
class Solution {
public:
    int dfs(map<int,int> &a,TreeNode*root){
        a[root->val]++;
        if(root->left == NULL and root->right == NULL){
            int b = 0;
            for(auto x : a)
            {
                if(x.second % 2 != 0) {
                   b++;
                    
                }
            }
            
             a[root->val]--;
            if(b < 2)
            return 1;
            return 0;
        }
        
        int ans = 0;
        if(root->right != NULL){
            ans += dfs(a,root->right);
        }
        
        if(root->left != NULL){
            ans += dfs(a,root->left);
        }
          a[root->val]--;
        return ans;
    }
    
    int pseudoPalindromicPaths (TreeNode* root) {
        map<int,int> mp;
        return dfs(mp,root);
    }
};

File: 1457-pseudo-palindromic-paths-in-a-binary-tree/1457-pseudo-palindromic-paths-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution 
{
    int ans = 0;
    public int pseudoPalindromicPaths (TreeNode root)
    {
        if(root != null)
        call(root,new int[10]);
        return ans;
    }
    
    void call(TreeNode r, int [] v)
    {
        if(r.left == null && r.right == null)
        {
            v[r.val]++;
            int g = 0; 
            for(int i = 1; i<=9; i++)
                if(v[i]%2 != 0)
                    g++;
            if(g<=1)
                ans++;
            v[r.val]--;
        }
        else
        {
            v[r.val]++;
            if(r.left != null)
            {
                call(r.left,v);
            }
            if(r.right != null)
            {
                call(r.right,v);
            }
            v[r.val]--;
        }
    }
}